client/my-sites/checkout/checkout/web-payment-box.jsx

```diff
@@ -28,523 +26,383 @@ import {
-const APPLE_PAY_MERCHANT_IDENTIFIER = config( 'apple_pay_merchant_id' );
 const PAYMENT_REQUEST_OPTIONS = {
 	requestPayerName: true,
 	requestPayerPhone: false,
 	requestPayerEmail: false,
 	requestShipping: false,
 };
 
-/**
- * The `<WebPaymentBox />` component.
- */
-export class WebPaymentBox extends React.Component {
-	static propTypes = {
-		cart: PropTypes.shape( {
-			currency: PropTypes.string.isRequired,
-			total_cost: PropTypes.number.isRequired,
-			products: PropTypes.array.isRequired,
-		} ).isRequired,
-		transaction: PropTypes.shape( {
-			payment: PropTypes.object,
-		} ).isRequired,
-		transactionStep: PropTypes.shape( {
-			name: PropTypes.string.isRequired,
-			error: PropTypes.object,
-		} ).isRequired,
-		countriesList: PropTypes.array.isRequired,
-		onSubmit: PropTypes.func.isRequired,
-		translate: PropTypes.func.isRequired,
-	};
-
-	state = { processorCountry: 'US' };
-
-	constructor() {
-		super();
-		this.detectedPaymentMethod = detectWebPaymentMethod();
-	}
-
-	/**
-	 * @return {object} A dictionary containing `default`, `disabled` and `text` keys.
-	 */
-	getButtonState = () => {
-		const { transactionStep, translate } = this.props;
-
-		const defaultState = () => {
-			return {
-				default: true,
-				disabled: false,
-				text: translate( 'Select a payment card', {
-					context: 'Loading state on /checkout',
-				} ),
-			};
-		};
-		const ongoingState = () => {
-			return {
-				default: false,
-				disabled: true,
-				text: translate( 'Sending your purchase', {
-					context: 'Loading state on /checkout',
-				} ),
-			};
-		};
-		const completingState = () => {
-			return {
-				default: false,
-				disabled: true,
-				text: translate( 'Completing your purchase', {
-					context: 'Loading state on /checkout',
-				} ),
-			};
-		};
-
-		switch ( transactionStep.name ) {
-			case BEFORE_SUBMIT:
-				return defaultState();
-
-			case INPUT_VALIDATION:
-				if ( transactionStep.error ) {
-					return defaultState();
-				}
-
-				return ongoingState();
-
-			case SUBMITTING_PAYMENT_KEY_REQUEST:
-				return ongoingState();
-
-			case RECEIVED_PAYMENT_KEY_RESPONSE:
-				if ( transactionStep.error ) {
-					return defaultState();
-				}
-
-				return ongoingState();
-
-			case SUBMITTING_WPCOM_REQUEST:
-			case REDIRECTING_FOR_AUTHORIZATION:
-				return completingState();
-
-			case RECEIVED_WPCOM_RESPONSE:
-				if ( transactionStep.error || ! transactionStep.data.success ) {
-					return defaultState();
-				}
-
-				return completingState();
-
-			default:
-				throw new Error( `Unknown transaction step: ${ transactionStep.name }.` );
+const WebPayButtonWithStripe = withStripe( WebPayButton );
+
+export function WebPaymentBox( {
+	cart,
+	countriesList,
+	onSubmit,
+	presaleChatAvailable,
+	translate,
+	disablePostalCodeDebounce,
+	children,
+} ) {
+	const paymentMethod = useMemo( () => detectWebPaymentMethod(), [] );
```

Comment:
Is the `useMemo` necessary here?  `detectWebPaymentMethod` should be fast (and it takes care of caching internally).

Reply:
Oh! I just read that [it can be expensive](https://github.com/Automattic/wp-calypso/blob/e5bf30394355680532797aa31de89d5835886f56/client/lib/web-payment/index.js#L31) but I didn't actually notice that it was doing the caching there. I can remove this too.

Reply:
Done.